From 884aefcbc370aee52d1a9d5196b6568f58b8dd2a Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Mon, 12 Nov 2018 14:17:38 +0800
Subject: [PATCH 5143/5242] MLK-20323: Enable 8 instaces for vpu decoder

commit  60709c8ae92f416b27e2a39bd8dba3cc8b03df54 from
https://source.codeaurora.org/external/imx/linux-imx.git

1. #define VPU_MAX_NUM_STREAMS 8
   #define VID_API_NUM_STREAMS 8
2. #define MAX_BUFFER_SIZE 0xc00000

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/mediasys_types.h |    4 ++--
 drivers/mxc/vpu-decoder-b0/vpu_b0.h         |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mxc/vpu-decoder-b0/mediasys_types.h b/drivers/mxc/vpu-decoder-b0/mediasys_types.h
index e2f2440..91e5fd2 100644
--- a/drivers/mxc/vpu-decoder-b0/mediasys_types.h
+++ b/drivers/mxc/vpu-decoder-b0/mediasys_types.h
@@ -61,8 +61,8 @@
 typedef unsigned int BOOL;
 #define FALSE 0
 #define TRUE 1
-#define VPU_MAX_NUM_STREAMS 3
-#define VID_API_NUM_STREAMS 4
+#define VPU_MAX_NUM_STREAMS 8
+#define VID_API_NUM_STREAMS 8
 #define VID_API_MAX_BUF_PER_STR 3
 #define VID_API_MAX_NUM_MVC_VIEWS 4
 #define MEDIAIP_MAX_NUM_MALONES 2
diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.h b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
index b9159cb..305afc9 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.h
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
@@ -50,7 +50,7 @@
 #define MMAP_BUF_TYPE_MASK 0xF0000000
 #define M0_BOOT_SIZE 0x1000000
 #define DCP_SIZE 0x3000000
-#define MAX_BUFFER_SIZE 0x800000
+#define MAX_BUFFER_SIZE 0xc00000
 #define UDATA_BUFFER_SIZE 0x1000
 #define SHARED_SIZE 0x00400000
 #define MAX_DCP_NUM 2
-- 
1.7.9.5

